### PR TITLE
Separate optional notam_random crate

### PR DESCRIPTION
**Pull request by tranzystorek-io**
Hi! This PR reorganizes the random utils a bit. While having a built-in rand is of course a nice batteries-included convenience, some users might want to switch to sth smaller like nanorand and they should have that option.

### CODE REVIEW

**Comment by Nazariglez** [comment]
Hey thank you so much for the PR, it makes sense to me. Let me check it a little and I'll come back to you.

File: Cargo.toml [hunk 1]
```diff
@@ -253,6 +255,10 @@ required-features = ["draw"]
 name = "log_basic"
 required-features = ["log"]
 
+[[example]]
+name = "render_instancing_cubes"
```
Comment (Nazariglez) [inline Cargo.toml:259]: This example's name is `renderer_instancing_cubes`

**Review by Nazariglez — CHANGES_REQUESTED** [review]
It looks good to me! If you can fix that minor typo we can merge it.

File: Cargo.toml (hunk 1 above)
Comment (tranzystorek-io) [reply Cargo.toml]: Done!

**Comment by Nazariglez** [comment]
Thanks!

**Pull request merged**